File: sephora-backend/Application/Dtos/Product/ProductDTO.cs
namespace CleanArchitecture.Application.Dtos.Product;

public class ProductDto
{
    public long Id { get; set; }
    public string Name { get; set; } = String.Empty;
    public string? Description { get; set; }
    public bool Active { get; set; }
    public BrandDto Brand { get; set; } = default!;
    public CategoryDto Category { get; set; }   
    public IEnumerable<ProductPieceDto>? Pieces { get; set; }
    public IEnumerable<RatingDto>? Ratings { get; set; }
    
    // TODO
    public decimal AverageRating { get; set; }
    
    public DateTime CreatedAt { get; set; }
    
    // Publication was less than 14 days ago
    public bool IsNew => Math.Abs((CreatedAt.Date - DateTime.UtcNow.Date).Days) < 14;
}

File: sephora-frontend/src/models/product/ProductDto.ts
import BrandDto from "../brand/BrandDto.ts";
import RatingDto from "../rating/RatingDto.ts";
import ProductPieceDto from "../piece/ProductPieceDto.ts";
import AmountDto from "../amount/AmountDto.ts";
import CategoryDto from "../category/CategoryDto.ts";

export default interface ProductDto {
    get id(): number;

    get name(): string | null;
    get description(): string;
    get active(): boolean;

    get brand(): BrandDto;
    get category(): CategoryDto;

    get pieces(): ProductPieceDto[];
    get ratings(): RatingDto[];
    get averageRating(): number;

    get volumes(): AmountDto[];

    get createdAt(): Date;
    get isNew(): boolean;
}
